src/resolvers/dockerfileTemplate.ts

```diff
@@ -0,0 +1,65 @@
+import * as Promise from 'bluebird'
+
+import * as DockerfileTemplate from 'dockerfile-template'
+
+import { Bundle, FileInfo, Resolver } from '../resolver'
+
+export default class DockerfileTemplateResolver implements Resolver {
+	public priority = 2
+	public name = 'Dockerfile.template'
+
+	private hasDockerfileTemplate = false
+	private templateContent: string
+	private bundle: Bundle
+
+	public entry(file: FileInfo): void {
+		if (file.name === 'Dockerfile.template') {
+			this.templateContent = file.contents
+			this.hasDockerfileTemplate = true
+		}
+	}
+
+	public provideBundle(bundle: Bundle): void {
+		this.bundle = bundle
+	}
+
+	public isSatisfied(): boolean {
+		return this.hasDockerfileTemplate
+	}
+
+	public resolve(): Promise<FileInfo[]> {
+		if (!this.hasDockerfileTemplate) {
+			throw new Error('Resolve called on resolver which is not satisfies')
+		}
+
+		const dockerfile: FileInfo = {
+			name: 'Dockerfile',
+			size: 0,
+			contents: ''
+		}
+
+		// Generate the variables to replace
+		const vars: DockerfileTemplate.TemplateVariables = {
+			RESIN_ARCH: this.bundle.architecture,
+			RESIN_MACHINE_NAME: this.bundle.deviceType
+		}
+
+		// Do the processing
+		try {
+			dockerfile.contents = DockerfileTemplate.process(this.templateContent, vars)
+		} catch(err) {
+			return Promise.reject(err)
+		}
```

Comment:
If you make this method async (`public async resolve(): Promise<FileInfo[]> { ...`), then you can instantly drop this try/catch wrapper. Throwing a synchronous error in here will return a rejected promise, returning a value will return a promise that resolves to that value. Magic.